spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoDataAutoConfiguration.java

```diff
@@ -113,6 +128,38 @@ public GridFsTemplate gridFsTemplate(MongoDbFactory mongoDbFactory,
 				mongoTemplate.getConverter());
 	}
 
+	@Nullable
+	private MongoDbFactory tryCreateDbFactoryForMongo(String database) {
+
+		MongoClient client = mongoClientOrNull(MongoClient.class);
+		if (client != null) {
+			return new SimpleMongoDbFactory(client, database);
+		}
+		return null;
+	}
+
+	@Nullable
+	private MongoDbFactory tryCreateDbFactoryForMongoClient(String database) {
+
+		com.mongodb.client.MongoClient client = mongoClientOrNull(
+				com.mongodb.client.MongoClient.class);
+		if (client != null) {
+			return new SimpleMongoClientDbFactory(client, database);
+		}
+		return null;
+	}
+
+	@Nullable
+	private <T> T mongoClientOrNull(Class<T> mongoClientType) {
+
+		try {
+			return this.beanFactory.getBean(mongoClientType);
```

Comment:
Rather than doing that, could you please inject an `ObjectProvider` and use its API. That would prevent from calling the `BeanFactory` directly.